backend/routes/recommendations.js

```diff
@@ -0,0 +1,64 @@
+const Parse = require('parse/node');
+const express = require("express")
+const router = express.Router()
+var cors = require('cors');
+
+router.use(cors())
+
+router.get('/:sessionToken', async (req, res) => {
+    //using the most recent positively reviewed book to create recs:
+    let query = new Parse.Query("_Session")
+    query.equalTo("sessionToken", req.params.sessionToken)
+    let userId = await query.first({useMasterKey : true})
+    userId = userId.attributes.user.id
+    let reviewQuery = new Parse.Query("Reviews")
+    reviewQuery.equalTo("userId", userId)
+    reviewQuery.greaterThan("rating", 3)
+    reviewQuery.descending("createdAt")
+    let bookSeed = await reviewQuery.first({useMasterKey : true})
+    let bookQuery = new Parse.Query("Books")
+    bookQuery.equalTo("bookId", bookSeed.attributes.bookId)
+    bookQuery.notEqualTo("userId", userId) //to get different user
+    bookQuery.equalTo("list", "Read")
+    let userLists = await bookQuery.find({useMasterKey : true})
+    let unfilteredRecs = userLists.map(async (u) => {
+        let userQuery = new Parse.Query("Books")
+        userQuery.equalTo("userId", u.attributes.userId)
+        userQuery.equalTo("list", "Read")
+        userQuery.notEqualTo("bookId", bookSeed.attributes.bookId)
+        let userQueryRes = await userQuery.find({useMasterKey : true})
+        return userQueryRes
+    })
+    unfilteredRecs = await Promise.all(unfilteredRecs)
+    unfilteredRecs = unfilteredRecs.flat()
+    //filter out the books already in the user's bookshelf
+    let filterQuery = new Parse.Query("Books")
+    filterQuery.equalTo("userId", userId)
+    const filterBooks = await filterQuery.find({useMasterKey: true})
+    const possibleRecs = unfilteredRecs.filter(rec => {
+        for (let i=0; i<filterBooks.length; i++) {
+            if (filterBooks[i].attributes.bookId===rec.attributes.bookId) {
+                return false
+            }
+        }
+        return true
+    })
+    let popularity = {}
```

Comment:
this is a good start. We can chat more in our 1-1 today but we probably want to add more signals/complexity here. In that case, it might be good to filter out large chunks of code into separate helper functions.